### ISSUE TITLE
`cast_possible_truncation` suggested fix replaces macro with definition

### ISSUE THREAD

**Issue by DanielEScherzer** [rust-lang/rust-clippy#17501]
### Summary

The suggested fix for `cast_possible_truncation` where the input comes from a macro replaces the macro with its definition - it shouldn't do that

### Reproducer

Code:

```rust
#![deny(clippy::cast_possible_truncation)]
macro_rules! get_i64 {
    () => { 5_i64 }
}

fn print_i32(v: i32) {
    println!("Value: {v}");
}

fn main() {
    print_i32(
        get_i64!() as i32
    );
}
```

Current output:
```
    Checking playground v0.0.1 (/playground)
error: casting `i64` to `i32` may truncate the value
  --> src/main.rs:12:9
   |
12 |         get_i64!() as i32
   |         ^^^^^^^^^^^^^^^^^
   |
   = help: if this is intentional allow the lint with `#[allow(clippy::cast_possible_truncation)]` ...
   = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.97.0/index.html#cast_possible_truncation
note: the lint level is defined here
  --> src/main.rs:1:9
   |
 1 | #![deny(clippy::cast_possible_truncation)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: ... or use `try_from` and handle the error accordingly
   |
12 -         get_i64!() as i32
12 +         i32::try_from(5_i64)
   |

error: could not compile `playground` (bin "playground") due to 1 previous error
```

Desired output:
```
    Checking playground v0.0.1 (/playground)
error: casting `i64` to `i32` may truncate the value
  --> src/main.rs:12:9
   |
12 |         get_i64!() as i32
   |         ^^^^^^^^^^^^^^^^^
   |
   = help: if this is intentional allow the lint with `#[allow(clippy::cast_possible_truncation)]` ...
   = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.97.0/index.html#cast_possible_truncation
note: the lint level is defined here
  --> src/main.rs:1:9
   |
 1 | #![deny(clippy::cast_possible_truncation)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: ... or use `try_from` and handle the error accordingly
   |
12 -         get_i64!() as i32
12 +         i32::try_from(get_i64!())
   |

error: could not compile `playground` (bin "playground") due to 1 previous error
```

### Version

```text
Playground 1.97.1
Clippy 0.1.97 (2026-07-14 8bab26f4f6)
```
